src/openforms/forms/api/serializers/logic/action_serializers.py

```diff
@@ -189,6 +192,20 @@ def validate(self, data: dict) -> dict:
                 code="blank",
             )
 
+        # validate format of value for date variable
+        if action_type and action_type == LogicActionTypes.variable:
+            # get related form variable from context of parent serializer
+            form_var = self.parent.parent.context["form_variables"].variables[variable]
```

Comment:
I believe the context is the same for the entire tree, meaning `self.context is self.parent.context is self.parent.parent.context`. This is also kinda required, since a serializer itself should not be concerned with how deeply it is nested (as that would break the ability to compose serializers).